tests/Drivers/DatabaseDriverTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/sonata project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\HelpersBundle\Tests\Drivers;
+
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Doctrine\ORM\EntityManager;
+use Lexik\Bundle\MaintenanceBundle\Drivers\Query\DsnQuery;
+use PHPUnit\Framework\TestCase;
+use Sonata\HelpersBundle\Drivers\DatabaseDriver;
+use Sonata\HelpersBundle\Drivers\Query\DefaultQuery;
+use Sonata\HelpersBundle\Tests\TestHelperTrait;
+use Sonata\PageBundle\Entity\SiteManager;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class DatabaseDriverTest.
+ *
+ * @author Quentin Belot <quentin.belot@ekino.com>
+ */
+class DatabaseDriverTest extends TestCase
+{
+    use TestHelperTrait;
+
+    /**
+     * @throws \ReflectionException
+     *
+     * @dataProvider setOptionsProvider
+     */
+    public function testSetOptions(array $options, string $class, bool $expectManager): void
+    {
+        $siteManager = $this->createMock(SiteManager::class);
+        $registry    = $this->createMock(Registry::class);
+        $driver      = new DatabaseDriver($siteManager, $registry);
+
+        if ($expectManager) {
+            $registry->expects($this->once())->method('getManager')->willReturn($this->createMock(EntityManager::class));
+        }
+
+        $driver->setOptions($options);
+
+        $this->assertInstanceOf($class, $this->getPrivatePropertyValue($driver, 'pdoDriver'));
+    }
+
+    public function setOptionsProvider(): \Generator
+    {
+        yield [
```

Comment:
yield should be named (will throw deprecation with phpunit8 and exception with phpunit9 if i well remember ;) )